pkg/kubelet/kubelet_pods.go

```diff
@@ -1528,7 +1528,7 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 			klog.V(4).Infof("Cannot get host IP: %v", err)
 		} else {
 			s.HostIP = hostIP.String()
-			if kubecontainer.IsHostNetworkPod(pod) && s.PodIP == "" {
+			if kubecontainer.IsHostNetworkPod(pod) {
```

Comment:
I guess the following is better as we don't want to update the Pod's IP even if it is not changed.

```suggestion
			if kubecontainer.IsHostNetworkPod(pod) && s.HostIP != s.PodIP {
```

Reply:
Make sense. Thanks a lot.